pulp_smash/tests/rpm/api_v2/test_remove_unit.py

```diff
@@ -0,0 +1,283 @@
+    }
+
+
+def _gen_distributor():
+    """Return a semi-random dict for use in creating a YUM distributor."""
+    return {
+        'auto_publish': False,
+        'distributor_id': utils.uuid4(),
+        'distributor_type_id': 'yum_distributor',
+        'distributor_config': {
+            'http': True,
+            'https': True,
+            'relative_url': utils.uuid4() + '/',
+        },
+    }
+
+
+class _BaseTestCase(unittest2.TestCase):
+    """Provide a server config, and tear down created resources."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Provide a server config and an iterable of resources to delete."""
+        cls.cfg = config.get_config()
+        cls.resources = set()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete created resources."""
+        client = api.Client(cls.cfg)
+        for resource in cls.resources:
+            client.delete(resource)
+
+
+class _CommonTestsMixin(object):
+    """Common tests for RemoveMissing* classes."""
+
+    def test_status_code(self):
+        """Verify th HTTP status code of each server response."""
+        for step, code in (
+                ('sync', 202),
+                ('publish', 202),
+                ('units before removal', 200),
+                ('units after removal', 200),
+        ):
+            with self.subTest(step=step):
+                for response in self.responses[step]:
+                    self.assertEqual(response.status_code, code)
+        for step, code in (
+                ('distribute', 201),
+                ('remove unit', 202),
+        ):
+            with self.subTest(step=step):
+                self.assertEqual(self.responses[step].status_code, code)
+
+    def test_task_error_traceback(self):
+        """Assert each task's "error" and "traceback" fields are null."""
+        for action in {'sync', 'remove unit'}:
+            for i, task in enumerate(self.task_bodies[action]):
+                for key in {'error', 'traceback'}:
+                    with self.subTest((i, key)):
+                        self.assertIsNone(task[key])
+
+    def test_task_progress_report(self):
+        """Assert no task's progress report contains error details."""
+        for i, task in enumerate(self.task_bodies['sync']):
+            with self.subTest(i=i):
+                self.assertEqual(
+                    task['progress_report']['yum_importer']['content']['error_details'],  # noqa pylint:disable=line-too-long
+                    []
+                )
+
+    def test_units_before_removal(self):
+        """Test that units in repositories before removal are the same."""
+        bodies = [re.json() for re in self.responses['units before removal']]
+        # Package category and package group will differ so we count only RPMs
+        self.assertEqual(
+            set(unit['unit_id'] for unit in bodies[0]
+                if unit['unit_type_id'] == 'rpm'),  # This test is fragile
```

Comment:
Please don't use hanging indents. This request is purely stylistic; but they're used nowhere in Pulp Smash right now.